### PR TITLE
OpenSSL 1.1.0 support

### LINKED ISSUE #4051: OpenSSL 1.1 not detected by configure
In Debian we've got reports from builds with OpenSSL 1.1.0 (beta?), where configure would fail:

```
Trying link with OPENSSL_LDFLAGS=; OPENSSL_LIBS=-lcrypto; OPENSSL_INCLUDES=
configure:18616: g++ -o conftest -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -DPACKAGEVERSION='"4.0.0~alpha3-1.Debian"' -std=gnu++11  -Wdate-time -D_FORTIFY_SOURCE=2 -fPIE -pie -Wl,-z,relro -Wl,-z,now -latomic   conftest.cpp -lcrypto  -lboost_thread -lboost_system -pthread  >&5
conftest.cpp: In function 'int main()':
conftest.cpp:38:17: error: too few arguments to function 'void CRYPTO_free(void*, const char*, int)'
 CRYPTO_free(NULL)
                 ^
In file included from conftest.cpp:34:0:
/usr/include/openssl/crypto.h:267:6: note: declared here
 void CRYPTO_free(void *ptr, const char *file, int line);
      ^
```

Please consider supporting OpenSSL 1.1.0 in 4.0.0.

### PR DESCRIPTION
**Pull request by zeha**
To fix #4051

### CODE REVIEW

**Comment by zeha** [comment]
All tests pass. Needs a tight review.

**Comment by rgacogne** [comment]
Looks good, very nice job!

File: m4/pdns_check_libcrypto.m4 [hunk 1]
```diff
@@ -1,62 +1,62 @@
...
 #     #include <openssl/hmac.h>
 #
 # LICENSE
 #
+# Taken and modified from AX_CHECK_OPENSSL by:
 #   Copyright (c) 2009,2010 Zmanda Inc. <http://www.zmanda.com/>
 #   Copyright (c) 2009,2010 Dustin J. Mitchell <dustin@zmanda.com>
 #
+# Modifications:
+#   Copyright (c) 2016 Pieter Lexis <pieter.lexis@powerdns.com>
```
Comment (Habbie) [inline m4/pdns_check_libcrypto.m4]: Please remove this.

Comment (Habbie) [inline m4/pdns_check_libcrypto.m4]: (And now that I notice it, please also remove it from systemd.m4 :) )

Comment (zeha) [inline m4/pdns_check_libcrypto.m4]: Added a fixup to @pieterlexis' commit, would squash before you merge this.

File: m4/pdns_check_libcrypto.m4 [hunk 2]
```diff
@@ -32,31 +29,31 @@
...
-            AC_PATH_PROG([PKG_CONFIG], [pkg-config])
+            AC_CHECK_TOOL([PKG_CONFIG], [pkg-config])
             if test x"$PKG_CONFIG" != x""; then
-                OPENSSL_LDFLAGS=`$PKG_CONFIG libcrypto --libs-only-L 2>/dev/null`
+                LIBCRYPTO_LDFLAGS=`$PKG_CONFIG openssl --libs-only-L 2>/dev/null`
                 if test $? = 0; then
-                    OPENSSL_LIBS=`$PKG_CONFIG libcrypto --libs-only-l 2>/dev/null`
-                    OPENSSL_INCLUDES=`$PKG_CONFIG libcrypto --cflags-only-I 2>/dev/null`
+                    LIBCRYPTO_LIBS='-lcrypto' # This *will* break if the linker name is not crypto
+                    LIBCRYPTO_INCLUDES=`$PKG_CONFIG openssl --cflags-only-I 2>/dev/null`
```
Comment (mind04) [inline m4/pdns_check_libcrypto.m4]: Why are you reverting to openssl.pc here (and above)? Is there no useful libcrypto.pc in openssl 1.1.0?

Comment (zeha) [inline m4/pdns_check_libcrypto.m4]: @pieterlexis could you comment on this?

Comment (pieterlexis) [inline m4/pdns_check_libcrypto.m4]: I don't know what I was thinking, patch incoming

File: m4/pdns_check_libcrypto.m4 [hunk 3]
```diff
@@ -92,35 +89,26 @@ AC_DEFUN([AX_CHECK_OPENSSL], [
         # it will just work!
     fi
 
+    # Don't pull in -lssl
+    LIBCRYPTO_LIBS='-lcrypto'
```
Comment (mind04) [inline m4/pdns_check_libcrypto.m4]: This can go, there is no -lssl if you use libcrypto.pc

**Pull request merged**